lib/ansible/module_utils/network/nxos/facts/vlans/vlans.py

```diff
@@ -47,24 +54,18 @@ def populate_facts(self, connection, ansible_facts, data=None):
         """
         objs = []
         if not data:
-            data = connection.get('show running-config | section ^vlan')
-        vlans = re.split(r'(,|-)', data.split()[1])
-        for v in vlans:
-            if not v.isdigit():
-                vlans.remove(v)
-
-        config = re.split(r'(^|\n)vlan', data)
-        for conf in config:
-            conf = conf.strip()
-            if conf:
-                if conf[0] in vlans:
-                    vlans.remove(conf[0])
-                    obj = self.render_config(self.generated_spec, conf)
-                    if obj and len(obj.keys()) > 1:
-                        objs.append(obj)
-
-        for v in vlans:
-            obj = self.render_config(self.generated_spec, v)
+            # Use structured for most of the vlan parameter states.
+            # This data is consistent across the supported nxos platforms.
+            structured = self.get_device_data(connection, 'show vlan | json')
```

Comment:
This will work when we support `httpapi`.  NXAPI does not support `| json`.  We should add a TBD to fix this when support is added.

Reply:
`| json` won't work for the devices that do have structured output support. We still have to support those devices as they are not End of Support yet according to Cisco website https://www.cisco.com/c/en/us/products/collateral/switches/nexus-7000-series-switches/eos-eol-notice-c51-742131.html

Reply:
@trishnaguha Hi Trishna. Mike and I are pushing back on this request. This module was very broken except for rudimentary test cases. It now works on _all_ of our regression testbeds and images, with a more comprehensive (and complex) set of test cases, with the exception of the 6.2 image you referenced. We would like to see this PR merged since it fixes so many issues, and open a new issue to address the legacy/EOL limitation separately. I added a TBD to the facts code with a suggested resolution for that. Thanks.

Reply:
cc @ikhan2010 @cidrblock @privateip